pkg/odo/cli/component/create.go

```diff
@@ -116,13 +116,16 @@ odo catalog list components
 %[1]s nodejs frontend --context ./frontend
 
 # Create new Java component with binary named sample.jar in './target' directory
-%[1]s java:8  --binary target/sample.jar
+%[1]s java:8 --s2i --binary target/sample.jar
```

Comment:
You could also add in the `--binary` flag description that it is S2I specific

Reply:
+1